core/src/main/scala/filodb.core/query/PartitionTimeRangeReader.scala

```diff
@@ -33,7 +35,23 @@ final class PartitionTimeRangeReader(part: ReadablePartition,
     def getDouble(columnNo: Int): Double = vectorIts(columnNo).asDoubleIt.next
     def getFloat(columnNo: Int): Float = ???
     def getString(columnNo: Int): String = ???
-    override def getHistogram(columnNo: Int): bv.Histogram = vectorIts(columnNo).asHistIt.next
+    override def getHistogram(columnNo: Int): bv.Histogram = {
+      try {
+        vectorIts(columnNo).asHistIt.next
+      } catch {
+        case e : EmptyHistogramException => {
+          var message = s"${e.getMessage} infos=["
+           while (infos.hasNext) {
+             val info = infos.nextInfo
+             message +=
+               s"""${info.debugString} """
+           }
+          message += "]"
+          logger.error(s"EmptyHistogramException message ${message}")
+          throw new IllegalArgumentException(message)
```

Comment:
What is the current exception thrown? Do you want to throw the entire message built (possibly large) or just log and throw a concise error? 

It would be good if you give the current problem, the exception seen and the expectation to create this PR in the PR description.

Reply:
Hi Amol,

here is the description. I updated it through the git message.

> 
> Some queries occasionally hit exceptions because of empty histogram.
> However, the same exception could not be reproduced later.
> The hunch is that the bug is caused by a race condition.
> So, adding additional debugging log to print out the chunk id chunk info and the memory dump.
>

Reply:
>Do you want to throw the entire message built (possibly large)

@amolnayak311 we met and discussed w/ @vishramachandran and @sherali42. The intent here is to dump everything to the logs that might be useful, but @yu-shipit  maybe we could add a `substr` call just to be doubly sure we don't log some ridiculous character count? (Full disclosure: I don't know what constitutes "reasonable." Pick something that agrees with local tests-- or maybe others have opinions about this already?)

Reply:
yup agree with @amolnayak311, no need to add debug log message to the exception. whatever we were throwing earlier `requirement failed: empty histogram` is good enough.

Reply:
I think we just add EmptyHistogramException to label the error message.
Thank you.

Reply:
@sherali42 @amolnayak311  we don't need the ChunkInfo string to debug this race condition?

Reply:
We could truncate this string too -- I'm not sure if that's a useful option, though.